### DISCUSSION TITLE
Add timeout argument to WebSocketTestSession.receive

### DISCUSSION THREAD

**Discussion by marcuslimdw** [encode/starlette#2595] [Ideas] [answered]
In my test, I'd like to fail if no message is received after a certain duration, but I can't do that easily, because currently, `WebSocketTestSession.receive` will wait (and block) forever. 

My suggestion is to add a `timeout: float | None = None` parameter to `receive`, `receive_text`, `receive_bytes`, and `receive_json`, which will eventually pass this argument to the underlying `Queue`.

**Comment by adriangb** [comment]
You can use anyio:

```python
with anyio.fail_after(123):
   await ws.receive()
```

**Reply by marcuslimdw** [reply]
I know that I can use anyio (or any other out-of-band timeout mechanism). I'm just wondering - since this is easily exposed in the API, why not?

**Reply by adriangb** [reply]
Because additional API surface area creates more code to test, maintain, etc. And it would be just doing the same thing internally most likely.

**Reply by marcuslimdw** [reply]
It would be using `queue.Queue`'s `timeout` argument, so I don't think so. Understood on the "won't fix" status.

**Comment by Kludex** [comment, marked as answer]
There's a previous issue about this. I suggest reading that.
